### PR TITLE
[Docs] Adding docs for developers and adding a new cloud

### PR DESCRIPTION
**Pull request by Michaelvll**
This PR adds the contributing.md in our docs and also include a guide for adding a new cloud for future reference.

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
  - [x] Rendered locally.
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: docs/requirements-docs.txt [hunk 1]
```diff
@@ -13,3 +13,4 @@ sphinxcontrib-jsmath==1.0.1
 sphinxcontrib-qthelp==1.0.3
 sphinxcontrib-serializinghtml==1.1.5
 sphinxemoji==0.2.0
+recommonmark==0.7.1
```
Comment (concretevitamin) [inline docs/requirements-docs.txt:16]: Cannot install on Python 3.7 and is deprecated: https://github.com/readthedocs/recommonmark?tab=readme-ov-file#recommonmark

Does the recommended [MyST](https://github.com/executablebooks/MyST-Parser) work?

**Review by concretevitamin — COMMENTED** [review]
Trying to build but hit an error on Py 3.7

File: docs/requirements-docs.txt (hunk 1 above)
Comment (Michaelvll) [reply docs/requirements-docs.txt]: It seems working for me with Python 3.10. Do we really need to support Python 3.7 for the docs only, considering many packages started deprecating the support of 3.7?

> Does the recommended [MyST](https://github.com/executablebooks/MyST-Parser) work?
MyST works as well, with a small cross-reference issue, where `smoke tests` fails to cross-reference the title `### Testing` due to the a problem with MyST, which can be ignored.

File: docs/source/index.rst [hunk 2]
```diff
@@ -158,3 +158,10 @@ Documentation
    reference/cli
    reference/api
    reference/config
+
+.. toctree::
```
Comment (concretevitamin) [inline docs/source/index.rst:162]: How about putting this under User Guides? 

(Not feeling strongly): docs' nav bar is getting a bit long; wdyt about making "Developer Guides" a dropdown? e.g., on click, show CONTRIBUTING / new-cloud.

File: docs/source/developers/new-cloud.rst
Comment (concretevitamin) [inline docs/source/developers/new-cloud.rst:1]: The UX of force-redirect is a bit surprising and doesn't encourage users to stay on our docs site.

How about in index.rst directly link to the gdocs? It'll show a little icon there to indicate external links.

Or keep this current rst, but no redirect and just use text to link to that doc?

**Review by concretevitamin — COMMENTED** [review]
Thanks @Michaelvll! I can build now.

File: docs/source/index.rst (hunk 2 above)
Comment (Michaelvll) [reply docs/source/index.rst:162]: Sounds good. Moved.
For the dropdown. It seems non-trivial to make a single header dropdown. I will probably keep the current expanded version for now.

File: docs/source/developers/new-cloud.rst
Comment (Michaelvll) [reply docs/source/developers/new-cloud.rst:1]: Changed to an external link. Weirdly, it did not for me before, but now it works fine...

File: docs/source/index.rst [hunk 3]
```diff
@@ -142,6 +142,13 @@ Documentation
    reference/logging
    reference/faq
 
+.. toctree::
+   :maxdepth: 0
+   :caption: Developer Guides
+
+   developers/CONTRIBUTING
+   Adding a New Cloud <https://docs.google.com/document/d/1oWox3qb3Kz3wXXSGg9ZJWwijoa99a3PIQUHBR8UgEGs/edit?usp=sharing>
```
Comment (concretevitamin) [inline docs/source/index.rst:150]:
```suggestion
   Guide: Adding a New Cloud <https://docs.google.com/document/d/1oWox3qb3Kz3wXXSGg9ZJWwijoa99a3PIQUHBR8UgEGs/edit?usp=sharing>
```

File: docs/source/index.rst (hunk 2 above)
Comment (concretevitamin) [reply docs/source/index.rst]: Sounds good. We could look into how to restructure the grouping of different sections; currently they feel a bit too long.

**Review by concretevitamin — APPROVED** [review]
LGTM, thanks @Michaelvll.

File: docs/source/index.rst (hunk 3 above)
Comment (Michaelvll) [reply docs/source/index.rst:150]: Good call! Added.

**Pull request merged**